src/main/java/programminglife/model/Graph.java

```diff
@@ -0,0 +1,127 @@
+package programminglife.model;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+/**
+ * Created by marti_000 on 25-4-2017.
+ */
+public class Graph {
+    private String id;
+    private Set<Node> rootNodes;
+    private List<Node> nodes;
+
+    public Graph(String id) {
+        this.nodes = new ArrayList<>();
+        this.nodes.add(new Node(-1));
+        this.id = id;
+        this.rootNodes = new HashSet<>();
+    }
+
+    public boolean addNode(Node node) {
+        return this.nodes.add(node);
+    }
+
+    public Node getNode(int id) {
+        try {
+            return this.nodes.get(id);
+        } catch (IndexOutOfBoundsException e) {
+            return null;
+        }
+    }
+
+    private void parseLink(Scanner sc) {
+        int sourceId = Integer.parseInt(sc.next());
+        sc.next();
+        int destinationId = Integer.parseInt(sc.next());
+        sc.next();
+        String overlap = sc.next();
+
+        Node sourceNode = this.getNode(sourceId);
+        Node destinationNode = this.getNode(destinationId);
+
+        if (null == sourceNode)
+            throw new Error("Source node does not (yet) exist!");
+
+        if (null == destinationNode) {
+            destinationNode = new Node(destinationId);
+            this.addNode(destinationNode);
+        }
+
+        sourceNode.addChild(destinationNode);
+        destinationNode.addParent(sourceNode);
+    }
+
+    public static Graph parse(String file) throws FileNotFoundException {
+        System.out.println(String.format("Parsing file %s", file));
+
+        Scanner lineScanner = new Scanner(new File(file));
+        Graph graph = new Graph(null);
+
+        int numLinesRead = 0;
+        int numNodesParsed = 0;
+        int numLinksParsed = 0;
+        int miscParsed = 0;
+
+        boolean verbose = false;
+
+        while (lineScanner.hasNextLine()) {
+            String line = lineScanner.nextLine();
+            Scanner tokenScanner = new Scanner(line);
+            tokenScanner.useDelimiter("\t");
+            String token = tokenScanner.next();
+            numLinesRead++;
+
+            if (numLinesRead % 2500 == 0)
+                verbose = true;
+            else
+                verbose = false;
+
+            if (verbose) System.out.println(String.format("Token %s read (line %d)... ", token, numLinesRead));
+
+            switch (token) {
+                case "S":
+                    // Parse segment
+//                    System.out.println("parsing segment");
+                    Node parsedNode = Node.parseSegment(tokenScanner);
```

Comment:
Why is there a method for parse link but not one for parse Segment. I would prefer parse segment to be in a seperate method.